app/models/concerns/orchestration/realm.rb

```diff
@@ -45,26 +45,31 @@ def update_realm
   private
 
   def queue_realm
-    return unless realm? && errors.empty?
+    return unless (realm? || (old && old.realm?)) && errors.empty?
```

Comment:
The complex conditions in `unless` statement are quite hard to reason about (at least for me). Would you mind expanding it using `if` instead?

Reply:
Sorry if I'm being stubborn here, but I think this way it's most straightforward way and consistent with other orchestration code. I can change it, but I think it won't improve the code.

Reply:
I usually agree that multiple conditions and `unless` is hard to reason about, but the `return if` version of this looks like it'd be more convoluted TBH.

Reply:
take it or leave it, I just know it's hard to read for me.